### PR TITLE
[VEN-963, VEN-967] Certik audit 

### PR DESCRIPTION
**Pull request by defcon022**
## Description
* 963 and 967 Certik audit fix.

Resolves #

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/Rewards/RewardsDistributor.sol [hunk 1]
```diff
@@ -256,9 +256,7 @@ contract RewardsDistributor is ExponentialNoError, Ownable2StepUpgradeable {
      * @param vTokens The list of markets to claim REWARD TOKEN in
      */
     function claimRewardToken(address holder, VToken[] memory vTokens) public {
-        address[] memory holders = new address[](1);
-        holders[0] = holder;
-        _claimRewardToken(holders, vTokens);
+        _claimRewardToken(holder, vTokens);
```
Comment (chechu) [inline contracts/Rewards/RewardsDistributor.sol:259]: Do we need the internal function? Would it make sense just move the function content of `_claimRewardToken` to this public function? We are not invoking `_claimRewardToken` from any other place

Comment (defcon022) [reply contracts/Rewards/RewardsDistributor.sol:259]: cool, will move.

Comment (defcon022) [reply contracts/Rewards/RewardsDistributor.sol]: [FIXED](https://github.com/VenusProtocol/isolated-pools/pull/163/commits/b3d53030012993270beaf7fa074763c64e5f7645)

**Review by 0xlucian — APPROVED** [review]

File: contracts/ComptrollerStorage.sol [hunk 2]
```diff
@@ -104,10 +104,34 @@ contract ComptrollerStorage {
...
+
     /**
      * @dev This empty reserved space is put in place to allow future versions to add new
      * variables without shifting down storage in the inheritance chain.
      * See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
      */
     uint256[50] private __gap;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(address poolRegistry_, address accessControl_) {
```
Comment (narayanprusty) [inline contracts/ComptrollerStorage.sol:127]: @defcon022 Shouldn't the constructor be in `Comptroller` contract instead of `ComptrollerStorage`?

Comment (defcon022) [reply contracts/ComptrollerStorage.sol:127]: `ComptrollerStorage` is inherited in `Comptroller` and every storage variable is been shifted to `ComptrollerStorage` contract. 

The same issue been raised by Certik.

**Review by kkirka — APPROVED** [review]

**Review by kkirka — COMMENTED** [review]
I personally don't think immutables belong in the storage contract. These are like constants that are put to code, not to the storage slots of the contract, so they are kinda like implementation. But I don't want to step in the way here.

**Pull request merged**